Kaguya/Kaguya/Discord/Commands/Configuration/AntiraidSetup.cs

```diff
@@ -0,0 +1,434 @@
+﻿using System;
+using System.Text;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+using Discord.WebSocket;
+using Humanizer;
+using Humanizer.Localisation;
+using Interactivity;
+using Interactivity.Confirmation;
+using Kaguya.Database.Model;
+using Kaguya.Database.Repositories;
+using Kaguya.Discord.DiscordExtensions;
+using Kaguya.Discord.Parsers;
+using Kaguya.Internal.Attributes;
+using Kaguya.Internal.Enums;
+using Microsoft.Extensions.Logging;
+
+namespace Kaguya.Discord.Commands.Configuration
+{
+    [Module(CommandModule.Configuration)]
+    [Group("antiraid")]    
+    [Alias("ar")]
+    [RequireUserPermission(GuildPermission.Administrator)]
+    [RequireBotPermission(GuildPermission.ManageRoles)]
+    [RequireBotPermission(GuildPermission.ManageChannels)]
+    [RequireBotPermission(GuildPermission.BanMembers)]
+    [RequireBotPermission(GuildPermission.KickMembers)]
+    [RequireBotPermission(GuildPermission.MuteMembers)]
+    [RequireBotPermission(GuildPermission.DeafenMembers)]
+    public class AntiraidSetup : KaguyaBase<AntiraidSetup>
+    {
+        private readonly ILogger<AntiraidSetup> _logger;
+        private readonly InteractivityService _interactivityService;
+        private readonly KaguyaServerRepository _kaguyaServerRepository;
+        private readonly AntiraidConfigRepository _antiraidConfigRepository;
+        private readonly CommonEmotes _commonEmotes;
+
+        public AntiraidSetup(ILogger<AntiraidSetup> logger, InteractivityService interactivityService,
+            KaguyaServerRepository kaguyaServerRepository, AntiraidConfigRepository antiraidConfigRepository, 
+            CommonEmotes commonEmotes) : base(logger)
+        {
+            _logger = logger;
+            _interactivityService = interactivityService;
+            _kaguyaServerRepository = kaguyaServerRepository;
+            _antiraidConfigRepository = antiraidConfigRepository;
+            _commonEmotes = commonEmotes;
+        }
+
+        [Command(RunMode = RunMode.Async)]
+        [Summary("Displays an interactive setup utility for configuring the Kaguya Antiraid service.")]
+        public async Task AntiraidSetupCommand()
+        {
+            /* todo: Ensure more than 1 setup cannot run in a given server at a time. */
```

Comment:
@BitMasher We need to somehow check to make sure the same server hasn't executed this command more than once while a current setup is running (i.e. before the "stages" below are finished). We would somehow cache the server id into a thread-safe collection somewhere in the program, then remove the id once the "stages" are complete. If the server id is in the collection, we send an error.

What is the best way to complete this?

Reply:
a concurrentdictionary to track setup state could work